jsettlers.graphics/src/main/java/jsettlers/graphics/map/controls/original/panel/content/settlers/profession/ProfessionPanel.java

```diff
@@ -69,9 +70,13 @@ public void update(IPartitionSettings partitionSettings) {
 			this.panel.setup(professionSettings);
 		}
 	}
+	
+	public void setPlayer(IInGamePlayer player) {
+		uiContentUpdater.setPlayer(player);
+	}
 
 	private static IPartitionSettings currentDistributionSettingsProvider(IGraphicsGrid grid, ShortPoint2D position) {
-		return grid.getPlayerIdAt(position.x, position.y) >= 0 ? grid.getPartitionData(position.x, position.y).getPartitionSettings() : null;
+		return grid.getPlayerAt(position.x, position.y).getPlayerId() >= 0 ? grid.getPartitionData(position.x, position.y).getPartitionSettings() : null;
```

Comment:
grid.getPlayerAt() might return null that is the equivalent if a playerId of -1 or lower.
.getPlayerId() will never return anything below 0 but will just throw a NullPointerException.